cytotable/sources.py

```diff
@@ -93,16 +95,22 @@ def _get_source_filepaths(
         )
         # ensure the subpaths meet certain specifications
         if (
-            targets is None
-            or targets == []
-            # checks for name of the file from targets (compartment + metadata names)
-            or str(subpath.stem).lower() in [target.lower() for target in targets]
-            # checks for sqlite extension (which may include compartment + metadata names)
-            or subpath.suffix.lower() == ".sqlite"
+            # If targets are specified, only include files matching targets
+            (
+                targets is not None
+                and str(subpath.stem).lower() in [target.lower() for target in targets]
+                or subpath.suffix.lower() == ".sqlite"
+            )
+            # Otherwise, include files matching the source_datatypes
+            or (
+                targets is None
+                or targets == []
+                and subpath.suffix.lower() in source_datatypes
+            )
```

Comment:
This is a lot clearer than before; kudos!

Reply:
Thanks, agreed!